extension/encoding/googlecloudlogentryencodingextension/internal/armorlog/parser.go

```diff
@@ -0,0 +1,349 @@
+	BackendTargetProjectNumber string                     `json:"backendTargetProjectNumber"`
+	RemoteIP                   string                     `json:"remoteIp"`
+	CacheDecision              []string                   `json:"cacheDecision"`
+	SecurityPolicyRequestData  *securityPolicyRequestData `json:"securityPolicyRequestData"`
+
+	// Security policy fields: exactly one must be non-nil.
+	EnforcedSecurityPolicy     *enforcedSecurityPolicy `json:"enforcedSecurityPolicy"`
+	PreviewSecurityPolicy      *securityPolicyExtended `json:"previewSecurityPolicy"`
+	EnforcedEdgeSecurityPolicy *securityPolicyBase     `json:"enforcedEdgeSecurityPolicy"`
+	PreviewEdgeSecurityPolicy  *securityPolicyBase     `json:"previewEdgeSecurityPolicy"`
+}
+
+type securityPolicyRequestData struct {
+	RecaptchaActionToken  *recaptchaToken `json:"recaptchaActionToken"`
+	RecaptchaSessionToken *recaptchaToken `json:"recaptchaSessionToken"`
+	UserIPInfo            *userIPInfo     `json:"userIpInfo"`
+	RemoteIPInfo          *remoteIPInfo   `json:"remoteIpInfo"`
+	TLSJa4Fingerprint     string          `json:"tlsJa4Fingerprint"`
+	TLSJa3Fingerprint     string          `json:"tlsJa3Fingerprint"`
+}
+
+type recaptchaToken struct {
+	Score float64 `json:"score"`
+}
+
+type userIPInfo struct {
+	Source    string `json:"source"`
+	IPAddress string `json:"ipAddress"`
+}
+
+type remoteIPInfo struct {
+	IPAddress  string `json:"ipAddress"`
+	RegionCode string `json:"regionCode"`
+	ASN        int    `json:"asn"`
+}
+
+type securityPolicyBase struct {
+	Name             string `json:"name"`
+	Priority         int    `json:"priority"`
+	ConfiguredAction string `json:"configuredAction"`
+	Outcome          string `json:"outcome"`
+}
+
+type securityPolicyExtended struct {
+	securityPolicyBase
+	RateLimitAction      *rateLimitAction    `json:"rateLimitAction"`
+	PreconfiguredExprIds []string            `json:"preconfiguredExprIds"`
+	ThreatIntelligence   *threatIntelligence `json:"threatIntelligence"`
+	AddressGroup         *addressGroup       `json:"addressGroup"`
+}
+
+type enforcedSecurityPolicy struct {
+	securityPolicyExtended
+	AdaptiveProtection *adaptiveProtection `json:"adaptiveProtection"`
+}
+
+type rateLimitAction struct {
+	Key     string `json:"key"`
+	Outcome string `json:"outcome"`
+}
+
+type adaptiveProtection struct {
+	AutoDeployAlertID string `json:"autoDeployAlertId"`
+}
+
+type threatIntelligence struct {
+	Categories []string `json:"categories"`
+}
+
+type addressGroup struct {
+	Names []string `json:"names"`
+}
+
+func isValid(log armorlog) error {
+	if log.Type != armorLogType {
+		return fmt.Errorf("expected @type to be %s, got %s", armorLogType, log.Type)
+	}
+
+	if log.StatusDetails != statusDetailsBodyDeniedBySecurityPolicy &&
+		log.StatusDetails != statusDetailsDeniedBySecurityPolicy &&
+		log.StatusDetails != statusDetailsRedirectedBySecurityPolicy {
+		return fmt.Errorf("invalid statusDetails for Cloud Armor log entry: %s", log.StatusDetails)
+	}
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/44192/files#r2517961009 (just because we also follow the same for AWS logs encoding)